packages/graphql/src/schema/validation/custom-rules/directives/coalesce.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type {
+    DirectiveNode,
+    ObjectTypeDefinitionNode,
+    FieldDefinitionNode,
+    EnumTypeDefinitionNode,
+    InterfaceTypeDefinitionNode,
+} from "graphql";
+import { Kind } from "graphql";
+import { sameTypeArgumentAsField } from "../utils/same-type-argument-as-field";
+import { getInnerTypeName, isArrayType } from "../utils/utils";
+import { GRAPHQL_BUILTIN_SCALAR_TYPES, isSpatial, isTemporal } from "../../../../constants";
+import { DocumentValidationError } from "../utils/document-validation-error";
+
+export function verifyCoalesce(enums?: EnumTypeDefinitionNode[]) {
+    return function ({
+        directiveNode,
+        traversedDef,
+    }: {
+        directiveNode: DirectiveNode;
+        traversedDef: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode | FieldDefinitionNode;
```

Comment:
There is a generic Rule that gets a DirectiveNode and checks the directive is valid. Directives from the library can be set at all these levels (object | interface | field). 
It depends on the specific Directive (eg coalesce) whether or not it can be applied on all or just some of them, so in my opinion all of them should be accepted and checked in each "specialised" validation function. Otherwise we would just be "lying" to Typescript, because nothing really prevents the validation logic to be executed even though it shouldn't. It can also lead to unexpected errors.
What do you think?

Reply:
I see - that makes sense, I just wanted to check!